app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/NotesApi.kt

```diff
@@ -0,0 +1,120 @@
+package de.westnordost.streetcomplete.data.osmnotes
+
+import de.westnordost.osmapi.OsmConnection
+import de.westnordost.osmapi.common.Handler
+import de.westnordost.osmapi.common.errors.*
+import de.westnordost.osmapi.notes.NotesDao
+import de.westnordost.osmapi.notes.QueryNotesFilters
+import de.westnordost.streetcomplete.data.osm.mapdata.BoundingBox
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+import de.westnordost.streetcomplete.data.osm.mapdata.toOsmApiBoundingBox
+
+
+/**
+ * Creates, comments, closes, reopens and search for notes.
+ * All interactions with this class require an OsmConnection with a logged in user.
+ *
+ * @see NotesDao which is internally used.
+ */
+class NotesApi(osm: OsmConnection) {
+    private val notesDao: NotesDao = NotesDao(osm)
+
+    /**
+     * Create a new note at the given location
+     *
+     * @param pos position of the note.
+     * @param text text for the new note. Must not be empty.
+     *
+     * @throws OsmAuthorizationException if this application is not authorized to write notes
+     *                                   (Permission.WRITE_NOTES)
+     *
+     * @return the new note
+     */
```

Comment:
Hmm right, this means that the exception classes (etc) need to be wrapped as well.